tc/autotuner/utils.cc

```diff
@@ -0,0 +1,206 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "tc/autotuner/utils.h"
+
+#include <algorithm>
+#include <sstream>
+
+#include <glog/stl_logging.h>
+
+#include "tc/core/cuda/cuda_compilation_cache.h"
+#include "tc/core/cuda/cuda_mapping_options.h"
+#include "tc/core/flags.h"
+#include "tc/core/utils/dlpack.h"
+#include "tc/core/utils/time.h"
+#include "tc/lang/canonicalize.h"
+
+namespace tc {
+namespace autotune {
+
+std::vector<std::size_t> powers2andCeilDivisors(std::size_t val) {
+  auto numPowers = static_cast<std::size_t>(std::ceil(std::log2(val)));
```

Comment:
This requires a "#include <cmath>"

Reply:
I added #include <cmath> if that is what you meant?